src/pypromice/process/L2toL3.py

```diff
@@ -315,72 +388,82 @@ def calcHumid(T_0, T_100, T_h, es_0, es_100, eps, p_h, RH_cor_h):
-    RH_cor : xarray.DataArray
-        Relative humidity corrected
-    z_boom : xarray.DataArray
-        Boom height
-    
+    df_in : pandas.Series
+        Series of observed latitude, longitude or elevation with datetime index.
+    breaks: list
+        List of timestamps of station relocation. First and last item should be
+        None so that they can be used in slice(breaks[i], breaks[i+1])
+        
     Returns
     -------
-    SHF : xarray.DataArray
-        Sensible heat flux corrected
-    LHF : xarray.DataArray
-        Latent heat flux corrected
+    np.ndarray
+        Smoothed and interpolated values corresponding to the input series.
     '''
-    HF_nan = np.isnan(p) | np.isnan(T) | np.isnan(Tsurf) \
-        | np.isnan(RH_cor) | np.isnan(WS) | np.isnan(z_boom)
-    SHF[HF_nan] = np.nan
-    LHF[HF_nan] = np.nan 
-    return SHF, LHF
-      
-def cleanSpHumid(q_h, T, Tsurf, p, RH_cor):
-    '''Find invalid specific humidity data values and replace with NaNs, 
-    based on air temperature, surface temperature, air pressure, 
-    and corrected relative humidity
+    df_all = pd.Series(dtype=float)  # Initialize an empty Series to gather all smoothed pieces
+    breaks = [None] + breaks + [None]
+    for i in range(len(breaks) - 1):
+        df = df_in.loc[slice(breaks[i], breaks[i+1])].copy()
+               
+        # Drop NaN values and calculate the number of segments based on valid data
+        df_valid = df.dropna()
+        if df_valid.shape[0] > 2:        
+            # Fit linear regression model to the valid data range
+            x = pd.to_numeric(df_valid.index).values.reshape(-1, 1)
+            y = df_valid.values.reshape(-1, 1)
+            
+            model = LinearRegression()
+            model.fit(x, y)
+            
+            # Predict using the model for the entire segment range
+            x_pred = pd.to_numeric(df.index).values.reshape(-1, 1)
+            
+            y_pred = model.predict(x_pred)
+            df =  pd.Series(y_pred.flatten(), index=df.index)
+
+        
+        # Update df_all with predicted values for the current segment
+        df_all = pd.concat([df_all, df])
     
-    Parameters
-    ----------
-    q_h : xarray.DataArray
-        Specific humidity
-    T : xarray.DataArray
-        Air temperature
-    Tsurf : xarray.DataArray
-        Surface temperature
-    p : xarray.DataArray
-        Air pressure
-    RH_cor : xarray.DataArray
-        Relative humidity corrected
+    # Fill internal gaps with linear interpolation
+    df_all = df_all.interpolate(method='linear', limit_area='inside')
     
-    Returns
-    -------
-    q_h : xarray.DataArray
-        Specific humidity corrected'''
-    q_nan = np.isnan(T) | np.isnan(RH_cor) | np.isnan(p) | np.isnan(Tsurf)
-    q_h[q_nan] = np.nan
-    return q_h
- 
+    # Extrapolate for timestamps before the first valid measurement
+    first_valid_6_months = slice(None, df_in.first_valid_index() + pd.to_timedelta('180D'))
```

Comment:
What if there has been a relocation within the 180 days or before the first valid index.

```python
first_valid_index = pd.to_datetime('2024-02-03')
first_relocation = pd.to_datetime('2024-02-23')
delta_time = min(
    pd.to_timedelta('180D'), 
    first_relocation - first_valid_index,
)
if delta_time.total_seconds() > 0:
    extrapolation_period = slice(None, first_valid_index + delta_time)
    # TODO: Do the extrapolation
else:
    logger.warning("No valid data for extrapolation")
```

Reply:
Isn't the indices with nan values already extrapolated above using the linear regression model?
`model.predict(x_pred)`? `x_pred` includes indices to both valid and invalid values.